File: CodigoFonte/PrefeituraCachoeiro/PrefeituraCachoeiro.Aplicacao/Dtos/Requisicoes/CriarContratoRequest.cs
using Microsoft.AspNetCore.Http;
using System.Diagnostics.CodeAnalysis;

namespace PrefeituraCachoeiro.Aplicacao.Dtos.Requisicoes
{
    [ExcludeFromCodeCoverage]
    public class CriarContratoRequest
    {
        public DateTime DataContrato { get; set; }
        public string NumeroContrato { get; set; }
        public int EmpresaId { get; set; }
        public int TipoContratacao { get; set; }
        public string Gerente { get; set; }
        public DateTime DataInicio { get; set; }
        public DateTime DataTermino { get; set; }
        public int PrefeituraId { get; set; }
        public List<IFormFile>? Arquivos { get; set; }
        public IFormFile ArquivoTemplate { get; set; }
    }
}

File: Frontend/PrefeituraCachoeiro.App/src/app/request/ContratoRequest/criarContratoRequest.ts
export class CriarContratoRequest {
    DataContrato: Date;
    NumeroContrato: string = "";
    EmpresaId: number;
    TipoContratacao: number;
    Gerente: string = "";
    GerenteId: number;
    DataInicio: Date;
    DataTermino: Date;
    PrefeituraId: number;
    Aditivo?: number;
    TipoAditivo?: number;
    DataAssinaturaAditivo?: Date;
    DataValidadeAditivo?: Date;
    Arquivos: File[] = [];
    ArquivoTemplate: File;
    IsContratoGlobal: boolean;
}

export class salvarDocumentoContratoRequest {
    IdContrato: number;
    Arquivos: File[] = [];
}

export class salvarDocumentoAditivoRequest {
    IdAditivo: number;
    Arquivos: File[] = [];
}
